### PR TITLE
FirebaseApp modularization

### PR DESCRIPTION
**Pull request by Feiyang1**

### CODE REVIEW

File: packages/app/test/util.ts [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+ */
+
+import { FirebaseService } from '@firebase/app-types/private';
+import { FirebaseApp } from '@firebase/app-types';
+import { ComponentType, Component } from '@firebase/component';
+
+export class TestService implements FirebaseService {
+  constructor(private app_: FirebaseApp, public instanceIdentifier?: string) {}
+
+  // TODO(koss): Shouldn't this just be an added method on
```
Comment (hsubox76) [inline packages/app/test/util.ts]: I think this comment is outdated

File: packages/app/test/firebaseApp.test.ts [hunk 2]
```diff
@@ -15,22 +15,16 @@
  * limitations under the License.
  */
```
Comment (hsubox76) [inline packages/app/test/firebaseApp.test.ts]: Should this file be renamed? I don't think it's testing the `firebaseApp.ts` file specifically and it has the same name as the test file in `next/`

File: packages/app/package.json [hunk 3]
```diff
@@ -22,7 +22,7 @@
     "type-check": "tsc -p . --noEmit",
     "test:browser": "karma start --single-run",
     "test:browser:debug": "karma start --browsers Chrome --auto-watch",
-    "test:node": "TS_NODE_FILES=true TS_NODE_CACHE=NO TS_NODE_COMPILER_OPTIONS='{\"module\":\"commonjs\"}' nyc --reporter lcovonly -- mocha test/**/*.test.* --opts ../../config/mocha.node.opts",
+    "test:node": "TS_NODE_FILES=true TS_NODE_CACHE=NO TS_NODE_COMPILER_OPTIONS='{\"module\":\"commonjs\"}' nyc --reporter lcovonly -- mocha test/**/*.test.* src/**/*.test.ts --opts ../../config/mocha.node.opts",
     "prepare": "yarn build"
   },
   "license": "Apache-2.0",
```
[the remark this answers is not in the document]
Comment (Feiyang1) [reply packages/app/package.json]: Yes, I added tests in `src/`. Tried `./**/*.test.*`, but it doesn't run tests in `src/`. Any idea?

File: packages/app/rollup.config.js [hunk 4]
```diff
@@ -123,3 +131,4 @@ const es2017Builds = [
 ];
 
 export default [...es5Builds, ...es2017Builds];
+// export default [...es2017Builds];
```
[the remark this answers is not in the document]
Comment (Feiyang1) [reply packages/app/rollup.config.js]: Done

File: packages/app/next/package.json [hunk 5]
```diff
@@ -0,0 +1,8 @@
+{
+    "name": "@firebase/app/next",
+    "main": "../dist/cjs/next.js",
+    "browser": "../dist/cjs/next.js",
+    "module": "../dist/esm5/next.js",
+    "esm2017": "../dist/esm2017/next.js",
+    "typings": "../ådist/packages/app/src/next/index.d.ts"
```
[the remark this answers is not in the document]
Comment (Feiyang1) [reply packages/app/next/package.json]: Corrected.

File: packages/app/test/util.ts (hunk 1 above)
Comment (Feiyang1) [reply packages/app/test/util.ts]: Deleted.

File: packages/app/test/firebaseApp.test.ts (hunk 2 above)
Comment (Feiyang1) [reply packages/app/test/firebaseApp.test.ts]: That's probably a good idea. Renamed to `firebaseCompat.test.ts`

File: packages/app/package.json (hunk 3 above)
[the remark this answers is not in the document]
Comment (hsubox76) [reply packages/app/package.json]: I think I put that comment on a line that is gone now, never mind!

**Pull request closed without merging**